### PR TITLE
Save TF Shape

### PR DESCRIPTION
**Pull request by DanShort12**
## Linked Issues

N/A

## Description

This PR supports saving the TF Coils builder shape (and any other shape builders) to a json file.

## Interface Changes

Adds `save_shape` to `ParameterisedShapeBuilder`.

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Review by CoronelBuendia — CHANGES_REQUESTED** [review]
I think this might need a rebase? I'm pretty sure a lot of this came in with #622

File: bluemira/builders/EUDEMO/tf_coils.py [hunk 1]
```diff
@@ -131,15 +134,38 @@ class TFCoilsBuilder(OptimisedShapeBuilder):
...
+    def _extract_config(self, build_config: BuildConfig):
+        super()._extract_config(build_config)
+
+        self._geom_path = build_config.get("geom_path", None)
+        if self._geom_path is None and self._runmode.name.lower() == "read":
+            raise BuilderError(
+                "Must supply geom_path in build_config when using 'read' mode."
+            )
+
+        if self._geom_path is not None and os.path.isdir(self._geom_path):
```
[the remark this answers is not in the document]
Comment (DanShort12) [reply bluemira/builders/EUDEMO/tf_coils.py]: Tweaked the validation to account for this.

**Comment by DanShort12** [comment]
Needs already committed commits to be removed.

**Comment by DanShort12** [comment]
Rebase should now have just the relevant files.

**Review by CoronelBuendia — APPROVED** [review]

**Pull request merged**